libkbfs/block_types.go

```diff
@@ -367,6 +367,36 @@ func (db *DirBlock) SetIndirectPtrInfo(i int, info BlockInfo) {
 	db.IPtrs[i].BlockInfo = info
 }
 
+func (db *DirBlock) totalPlainSizeEstimate(
+	plainSize int, bsplit BlockSplitter) int {
+	if !db.IsIndirect() || len(db.IPtrs) == 0 {
+		return plainSize
+	}
+
+	// If the top block is indirect, it's too costly to estimate the
+	// sizes by checking the plain sizes of all the leafs.  Instead
+	// use the following imperfect heuristics:
+	//
+	// * If there are N child pointers, and the first one is a direct
+	//   pointer, assume N-1 of them are full.
+	//
+	// * If there are N child pointers, and the first one is an
+	//   indirect pointer, just give up and max out at the maximum
+	//   number of indirect pointers in a block, assuming that at
+	//   least one indirect block is full of pointers when there are
+	//   at least 2 indirect levels in the tree.
+	//
+	// This isn't great since it overestimates in many cases
+	// (especially when removing entries), and can vastly unerestimate
+	// if there are more than 2 levels of indirection.  But it seems
```

Comment:
Might be overcomplicated, but we could cache the depth of the tree at each level.

Reply:
I did think about putting the depth into the pointer itself, but that means that when we add a new level at the top of the tree, we'll have to fetch and update all the pointers for the entire tree, which seems like it would be too complicated and costly for something (a dir with more than two levels) that will probably be used almost never, and even then almost no one will ever care about the reported size of the directory in bytes.

Reply:
If you track the descending depth then you wouldn't need to do that.

Reply:
Oh, duh, of course.  Not sure what I was thinking.  I'll file a ticket, though I still don't think it's crucial enough to worry about right now.

Reply:
Agreed, thanks!